service-registry/src/main/java/org/apache/servicecomb/serviceregistry/discovery/AbstractDiscoveryFilter.java

```diff
@@ -30,6 +30,9 @@ public DiscoveryTreeNode discovery(DiscoveryContext context, DiscoveryTreeNode p
     }
 
     String childName = findChildName(context, parent);
+    if (childName == null) {
+      return null;
```

Comment:
when will this happened?
after return null, DiscoveryTree will throw exception:

```
// impossible, throw exception to help fix bug
throw new ServiceCombException(filter.getClass().getName() + " discovery return null.");
```